lib/Dialect/TritonGPU/IR/LayoutUtility.cpp

```diff
@@ -21,6 +22,20 @@ CTALayoutAttr permuteCTALayout(MLIRContext *ctx, CTALayoutAttr layout,
 
 LinearLayout getPaddedRegToSharedLayout(const LinearLayout &regLayout,
                                         PaddedSharedEncodingAttr paddedEnc) {
+  if (paddedEnc.getLinearComponent().has_value()) {
+    auto sharedLL = *paddedEnc.getLinearComponent();
+    auto regOutDims = regLayout.getOutDims();
+    // Ensure shared layout agrees with reg layout out dimensions
+    llvm::SmallDenseMap<StringAttr, int64_t> namedShape;
+    for (auto [name, size] : regLayout.getOutDims()) {
+      namedShape[name] = size;
+    }
+    sharedLL = ensureLayoutNotLargerThan(sharedLL, namedShape, "offset");
+    sharedLL = ensureLayoutNotSmallerThan(sharedLL, namedShape);
+
+    return regLayout.invertAndCompose(sharedLL);
```

Comment:
```suggestion
    auto sharedLL = *paddedEnc.getLinearComponent();
    return regLayout.invertAndCompose(sharedLL);
```
Doesn't this work? Note that `invertAndCompose` supports a larger codomain on the layout that you are going to invert. Even more, calling `ensureLayoutNotLargerThan` produces an incorrect result if you have to trim bases that are swizzled (this would not be the case here now, but it may be a usecaes in the future).